nodejs/awsx/ec2/vpc.ts

```diff
@@ -702,10 +709,15 @@ export interface VpcArgs {
     subnets?: VpcSubnetArgs[];
 
     /**
-     * The maximum number of availability zones to use in the current region.  Defaults to `2` if
-     * unspecified.  Use `"all"` to use all the availability zones in the current region.
+     * The names of the availability zones to use in the current region. Defaults to `2` if
+     * unspecified. Use `"all"` to use all the availability zones in the current region.
+     */
+    requestedAvailabilityZones?: number | "all" | [string, ...string[]] | pulumi.Input<string[]>;
+
+    /**
+     * @deprecated Use `requestedAvailabilityZones`
```

Comment:
One last comment I realized after I clicked 'approve' -- I think we want to avoid `@deprecated` here and make this a comment to avoid annoying the vast majority of users who've provided this parameter.

Reply:
@leezen Are you sure about this? 

I even think removing it instead of deprecating it would have been better. I mean this is one the benefits of using TS instead of Python and Pulumi should embrace these things instead of polluting the API.
The benefits is that the documentation and code becomes less dense and offers a path to upgrade your code.

Allow me to make the case that removing or even deprecating has tangible benefits. 
Think of it this way, if you neither remove or deprecate it, the users will *never* figure out that they're using deprecated or "outdated" implementations. The only other approach is for someone to examine each changelog for various libs and manage that themselves.

Deprecation allows us to warn that something has changed, and a compile error will tell them that the previous method was improper. 

Are you really sure you want me to remove it?

Reply:
Agree with pretty much everything you wrote, except that there's no user benefit to changing from one property to the other. The functionality will remain the same. So, we end up burdening users with additional cognitive overhead for something that isn't materially 'wrong' (there's no security issue, no incorrect functionality, etc.). Given it's likely to be a widely used parameter, I'm inclined to not mark it `@deprecated` until we cut a major version, at which point, we can simply drop the parameter completely as a hard breaking change.

Reply:
> So, we end up burdening users with additional cognitive overhead for something that isn't materially 'wrong' (there's no security issue, no incorrect functionality, etc.). 

I'm convinced, thanks, it makes sense! 

> Given it's likely to be a widely used parameter, I'm inclined to not mark it `@deprecated` until we cut a major version, at which point, we can simply drop the parameter completely as a hard breaking change.

I am not sure how the TS compiler works for straight up removing it in terms of offering an alternative, maybe add deprecated before released a new major version for a transition period, either way you're right.